MeshLib/MeshQuality/MeshValidation.cpp

```diff
@@ -147,4 +150,49 @@ MeshValidation::ElementErrorCodeOutput(const std::vector<ElementErrorCode> &erro
 	return output;
 }
 
+unsigned MeshValidation::detectHoles(MeshLib::Mesh const& mesh)
+{
+	MeshLib::Mesh* boundary_mesh (MeshSurfaceExtraction::getMeshBoundary(mesh));
+	std::vector<MeshLib::Element*> const& elements (boundary_mesh->getElements());
+
+	std::vector<unsigned> sfc_idx (elements.size(), std::numeric_limits<unsigned>::max());
+	unsigned current_surface_id (0);
+	std::vector<unsigned>::const_iterator it = sfc_idx.cbegin();
+
+	while (it != sfc_idx.cend())
+	{
+		int const idx = static_cast<int>(std::distance(sfc_idx.cbegin(), it));
```

Comment:
i am not sure in which case you get `idx<0` and `idx>elements.size()`. It looks for me that `idx<0` is impossible because `it` can take a value only in a range of [begin, end). you may get `idx>elements.size()` if `it==sfc_idx.cend()`, but the condition should be checked before calling distance().